lib/pages/LoginForm.dart

```diff
@@ -0,0 +1,191 @@
+import 'dart:collection';
+import 'dart:convert';
+
+import 'package:convert/convert.dart';
+import 'package:cryptography/cryptography.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:hive/hive.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'SplashScreen.dart';
+import 'HomePages.dart';
+
+class LoginLoadingPage extends StatefulWidget {
+  final String pin;
+
+  const LoginLoadingPage({Key? key, required this.pin}) : super(key: key);
+
+  @override
+  State<LoginLoadingPage> createState() => _LoginLoadingPageState();
+}
+
+class _LoginLoadingPageState extends State<LoginLoadingPage> {
```

Comment:
Since we're using this same three-option structure in the main.dart as well, is there any way that we can abstract this away so we don't have to write two new classes any time we want to wait on the splash screen for something?

Reply:
Hmm I'm not sure if there's a way to abstract this without introducing even more complicated code. The LoginLoadingBlahBlah's future returns either Error or a Hive Box, but the FutureBuilder in main is expecting a true/false, so the structure of our builder is going to be different. I'll sleep on it though

Reply:
Resolved by [0f0fe3e](https://github.com/Yours-App/Secure-Period-Tracker/pull/10/commits/0f0fe3ee301d88cd6ee8103d5a7835c319cd6163)